### PR TITLE
Add a Datadog build step

### PR DESCRIPTION
**Pull request by FlorianVeaux**
Adding a pipeline compatible build steps in order to:
- Provide configuration at the build level (atm: enabling log collection)
- Attach tags from environment variables that are only accessible from within the pipeline. (Example use case https://github.com/jenkinsci/datadog-plugin/issues/22)

### CODE REVIEW

File: src/test/java/org/datadog/jenkins/plugins/datadog/steps/DatadogOptionsTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    private static DatadogClientStub stubClient = new DatadogClientStub();
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        ClientFactory.setTestClient(stubClient);
+        DatadogGlobalConfiguration cfg = DatadogUtilities.getDatadogGlobalDescriptor();
+        ExtensionList.clearLegacyInstances();
+        cfg.setCollectBuildLogs(false);
+        EnvVars env = new EnvVars();
+        env.put("DD_TEST", "bar");
```
Comment (sarah-witt) [inline src/test/java/org/datadog/jenkins/plugins/datadog/steps/DatadogOptionsTest.java:33]: What is this for? Shouldn't it not be needed for the integration test?

File: src/test/java/org/datadog/jenkins/plugins/datadog/steps/DatadogOptionsTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        String[] expectedTags = new String[]{
+                "jenkins_url:" + DatadogUtilities.getJenkinsUrl(),
+                "user_id:anonymous",
+                "job:testMetricTags",
+                "result:SUCCESS",
+                "foo:bar"
+        };
+        stubClient.assertMetric("jenkins.job.duration", hostname, expectedTags);
+    }
+}
```
Comment (sarah-witt) [inline src/test/java/org/datadog/jenkins/plugins/datadog/steps/DatadogOptionsTest.java:70]: Could you add a test for multiple tags and setting `collectLogs: false`?

File: src/main/java/org/datadog/jenkins/plugins/datadog/steps/DatadogOptions.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+import org.jenkinsci.plugins.workflow.steps.BodyInvoker;
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+/**
+ * Pipeline plug-in step for recording time-stamps.
```
Comment (sarah-witt) [inline src/main/java/org/datadog/jenkins/plugins/datadog/steps/DatadogOptions.java:29]: This doesn't seem accurate

**Comment by stpierre** [comment]
Is there any documentation on how to use this? I tried copying the examples from the unit tests but I get `WorkflowScript: 27: Invalid option type "datadog". Valid option types: [...]`.

Happy to open an issue if that's the right course of action here.

**Comment by FlorianVeaux** [comment]
Hey @stpierre 
This has not been released yet, but we expect to release it sometime this week or the week after.
This page will get updated once done: https://plugins.jenkins.io/datadog/

**Pull request merged**